### ISSUE TITLE
security: test fails with go 1.10beta2

### ISSUE THREAD

**Issue by jmuk** [istio/istio#2818]
I've checked the tests with go 1.10beta2. Tests in security module fail:

```
--- FAIL: TestAwsGetInstanceIdentityDocument (0.00s)
        aws_test.go:120: Good Identity: Unexpected Error: failed to verify PKCS7 signature: x509: signature algorithm specifies an RSA public key, but have public key of type *dsa.PublicKey
--- FAIL: TestGetGetAgentCredential (0.00s)
        aws_test.go:223: Good Identity: Unexpected Error: failed to get EC2 instance identity document: failed to verify PKCS7 signature: x509: signature algorithm specifies an RSA public key, but have public key of type *dsa.PublicKey
FAIL
FAIL    istio.io/istio/security/pkg/platform    0.379s
```

**Comment by jmuk** [comment]
Checked a bit, seemingly the error message is generated at https://go.googlesource.com/go/+/master/src/crypto/x509/x509.go#907.

Indeed, https://github.com/fullsailor/pkcs7/blob/master/pkcs7.go#L257 looks like always passing `SHA1WithRSA` regardless of the certificate, and probably the data `AWSCertificatePem` does not match with this.

**Comment by jmuk** [comment]
cc @wattli @mangchiandjjoe

**Comment by jmuk** [comment]
seems fixed by #3250
